lib/diagnostic.js

```diff
@@ -5,17 +5,27 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (originalArray, callback, initalValue) {
+  const workingArray = originalArray.slice();
+  let accumulator = initalValue;
+  if (typeOf === 'undefined') {
+    accumulator = workingArray.shift();
+  }
+  workingArray.forEach(function (element, index /*, workingArray  */) {
+    accumulator = callback(accumulator, element, index, originalArray);
+  });
 };
 
+// this is not right but I think the answer is supposed to be similar to what we worked on yesterday with reduce
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
-const sum = function (/* previous, current */) {
-
+const sum = function (originalArray, callback, initalValue) {
+  let workingArray = originalArray.reduce(function(originalArray, callback, initalValue));
```

Comment:
You have an anonymous function declaration here but no surround braces to contain its body and this will break the entire script.